code/exercises/affine-cipher/test.ss

```diff
@@ -0,0 +1,29 @@
+(define (parse-test test)
+  (let* ((proc (string->symbol (lookup 'property test)))
+         (input (lookup 'input test))
+         (key (lookup 'key input))
+         (phrase (lookup 'phrase input))
+         (expected (lookup 'expected test)))
+    (if (string? expected)
+        `(test-success ,(lookup 'description test) equal? ,proc
+                       '(,key ,phrase) ,expected)
+        `(test-error ,(lookup 'description test) ,proc '(,key ,phrase)))))
+
+(define (spec->tests spec)
+  (let* ((cases (lookup 'cases spec))
+         (encoding (lookup 'cases (car cases)))
+         (decoding (lookup 'cases (cadr cases))))
+    (map parse-test (append encoding decoding))))
+
+(let ([spec (get-test-specification 'affine-cipher)])
+  (put-problem!
+    'affine-cipher
+    `((test . ,(spec->tests spec))
+       (version . ,(lookup 'version spec))
+       (skeleton . ,"affine-cipher.scm")
+       (solution . ,"example.scm")
+       (stubs encode decode)
+       (markdown
+        .
+        ,(splice-exercism 'affine-cipher
+                          '(sentence "Association lists will probably be your friend in this exercise if you want to avoid the complex computation required for decryption."))))))
```

Comment:
I didn't want to give too much away, but also math isn't super fun in `lisp`s

Reply:
I don't think it makes much sense to highlight particular data structures in descriptions, best to leave it open ended. I think the main purpose of adding additional writing to the readme is if the scheme version has some particularities or if default description is misleading.

Reply:
i'm not sure i agree... math isn't super fun not in lisp

Reply:
Sensible. I am not too attached to the comment.

Are you saying you think math is super fun in `lisp`s?

Reply:
I admit I like that we get to ignore the order of operations in favor of Scheme's order of evaluation.  That is quite convenient ;)

Reply:
check out [https://github.com/fedeinthemix/chez-scmutils/blob/master/scmutils/base.sls](https://github.com/fedeinthemix/chez-scmutils/blob/master/scmutils/base.sls)

Reply:
That is a wild library!